--- lib/system_check/gitlab_shell_check.rb.orig 2024-09-18 17:20:59.870390972 +0200
+++ lib/system_check/gitlab_shell_check.rb  2024-09-18 17:23:20.034139209 +0200
@@ -26,7 +26,7 @@
 
     def check_gitlab_shell_self_test
       gitlab_shell_repo_base = gitlab_shell_path
-      check_cmd = File.expand_path('bin/check', gitlab_shell_repo_base)
+      check_cmd = File.expand_path('bin/gitlab-shell-check', gitlab_shell_repo_base)
       $stdout.puts "Running #{check_cmd}"
 
       if system(check_cmd, chdir: gitlab_shell_repo_base)
