--- lib/tasks/gitlab/check.rake.orig	2013-05-16 13:52:25.780000000 +0200
+++ lib/tasks/gitlab/check.rake	2013-05-16 13:53:49.290000000 +0200
@@ -20,8 +20,6 @@
       check_gitlab_config_not_outdated
       check_log_writable
       check_tmp_writable
-      check_init_script_exists
-      check_init_script_up_to_date
       check_satellites_exist
 
       finished_checking "GitLab"
@@ -534,7 +532,7 @@
     ########################
 
     def gitlab_shell_user_home
-      File.expand_path("~#{Gitlab.config.gitlab_shell.ssh_user}")
+      File.dirname(File.dirname(File.expand_path("#{Gitlab.config.gitlab_shell.hooks_path}")))
     end
 
     def gitlab_shell_version
@@ -637,10 +635,10 @@
 
   def check_gitlab_shell
     print "GitLab Shell version? ... "
-    if gitlab_shell_version.strip == '1.1.0'
-      puts 'OK (1.1.0)'.green
+    if gitlab_shell_version.strip == '1.4.0'
+      puts 'OK (1.4.0)'.green
     else
-      puts 'FAIL. Please update gitlab-shell to v1.1.0'.red
+      puts 'FAIL. Please update gitlab-shell to v1.4.0'.red
     end
   end
 end
