--- lib/ci/api/entities.rb.orig	2017-07-24 14:14:27.234302357 +0200
+++ lib/ci/api/entities.rb	2017-07-24 14:14:44.463304303 +0200
@@ -52,12 +52,14 @@
           # when old API will be removed (planned for August 2017).
           model.options.dup.tap do |options|
             options[:image] = options[:image][:name] if options[:image].is_a?(Hash)
-            options[:services].map! do |service|
-              if service.is_a?(Hash)
-                service[:name]
-              else
-                service
-              end
+            if options[:services].respond_to? :map
+	      options[:services].map! do |service|	      
+                if service.is_a?(Hash)
+                  service[:name]
+                else
+                  service
+                end
+	      end
             end
           end
         end
