--- lib/ci/api/entities.rb.orig	2017-07-24 14:33:30.349724073 +0200
+++ lib/ci/api/entities.rb	2017-07-24 14:38:12.491003332 +0200
@@ -52,11 +52,13 @@
           # when old API will be removed (planned for August 2017).
           model.options.dup.tap do |options|
             options[:image] = options[:image][:name] if options[:image].is_a?(Hash)
-            options[:services].map! do |service|
-              if service.is_a?(Hash)
-                service[:name]
-              else
-                service
+            if options[:services].respond_to? :map
+              options[:services]&.map! do |service|
+                if service.is_a?(Hash)
+                  service[:name]
+                else
+                  service
+                end
               end
             end
           end
